quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -296,86 +304,78 @@ impl Source for KafkaSource {
                         None => {}
                     }
                 },
-                kafka_res = message_stream.next() => {
-                    if let Some(message_res) = kafka_res {
-                        let message = match message_res {
-                            Ok(message) => message,
-                            Err(KafkaError::PartitionEOF(partition_id)) => {
-                                self.state.num_active_partitions -= 1;
-                                info!(
-                                    topic = %self.topic,
-                                    partition_id = ?partition_id,
-                                    num_active_partitions = ?self.state.num_active_partitions,
-                                    "Reached end of partition."
-                                );
-                                continue;
-                            }
-                            // FIXME: This is assuming that Kafka errors are not recoverable, it may not be the
-                            // case.
-                            Err(err) => return Err(ActorExitStatus::from(anyhow::anyhow!(err))),
-                        };
-                        if let Some(doc) = parse_message_payload(&message) {
-                            docs.push(doc);
-                        } else {
-                            self.state.num_invalid_messages += 1;
+                message_res = self.consumer.recv() => {
```

Comment:
We get the message directly instead of an `Option<Message>` and remove one indentation level.